roles/migrationcontroller/tasks/mcg.yml

```diff
@@ -0,0 +1,101 @@
+- name: Checking if NooBaa CRDs exist
+  k8s_facts:
+    api_version: "apiextensions.k8s.io/v1beta1"
+    kind: CustomResourceDefinition
+    name: "{{ item }}"
+  loop: "{{ noobaa_expected_crds }}"
+  register: noobaa_found_crds
+
+- set_fact: 
+    noobaa_crds_found: "{{ noobaa_found_crds.results|length == noobaa_expected_crds|length }}"
+
+# Discover existing NooBaa System or create new 
+- when: noobaa_crds_found|bool
+  block: 
+  - name: Discover existing NooBaa system
+    k8s_facts: 
+      api_version: "noobaa.io/v1alpha1"
+      kind: NooBaa
+    register: noobaa_system
+
+  - set_fact:
+      noobaa_system_found: "{{ noobaa_system.resources|length > 0 }}"
+
+  - set_fact:
+      noobaa_system_namespace: "{{ mig_namespace if not noobaa_system_found else noobaa_system.resources[0].metadata.namespace }}"
+
+  - name: Creating a new NooBaa system
+    when: not noobaa_system_found|bool
+    k8s:
+     state: present
+     definition: "{{ lookup('template', 'noobaa_system.yml.j2') }}"
+
+# discover routes and secrets
+- block:
```

Comment:
The logic up to this point looks to me like:
1. Check if the CRD's are found
1. If they are found check for a noobaa system
1. If a nooba system is not found create one
1. Then we descend into this block regardless.

Won't tasks here fail if no noobaa system exists or can be created for lack of CRD's? Should this have at least a  `when: noobaa_crds_found|bool` conditional?

Reply:
@jmontleon yeah you're right. I didn't account for the use-case where none of the first two tasks fail